### PR TITLE
NGEN - Refactor rejit handler to remove inliners on module unload.

### PR DESCRIPTION
**Pull request by tonyredondo**
Possibly related to #2599

This PR refactors the rejit handler and corprofiler file to ensure the removal of unloaded modules from the inliners list.

### CODE REVIEW

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2602 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`WriteAndFlushEnrichedTraces`|net472|897μs|3.75μs|16.4μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|684μs|3.08μs|11.5μs|0| 0|0|2.57&nbsp;KB|
|#2602|`WriteAndFlushEnrichedTraces`|net472|900μs|4.33μs|17.3μs|0| 0|0|3.16&nbsp;KB|
|#2602|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|688μs|1.99μs|7.46μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendRequest`|netcoreapp3.1|288μs|779ns|3.02μs|0.143| 0|0|19.74&nbsp;KB|
|#2602|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2602|`SendRequest`|netcoreapp3.1|292μs|1.41μs|5.64μs|0.292| 0|0|19.74&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`ExecuteNonQuery`|net472|1.71μs|5.36ns|20ns|0.0942| 0|0|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`ExecuteNonQuery`|netcoreapp3.1|1.45μs|6.46ns|24.2ns|0.00891| 0|0|632&nbsp;B|
|#2602|`ExecuteNonQuery`|net472|1.81μs|7.5ns|27ns|0.0938| 0|0|594&nbsp;B|
|#2602|`ExecuteNonQuery`|netcoreapp3.1|1.53μs|8.2ns|45.6ns|0.00824| 0|0|632&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`CallElasticsearch`|net472|2.73μs|14.8ns|85.2ns|0.126| 0|0|803&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`CallElasticsearch`|netcoreapp3.1|1.64μs|9.19ns|56.7ns|0.0111| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`CallElasticsearchAsync`|net472|2.97μs|17ns|124ns|0.146| 0|0|939&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`CallElasticsearchAsync`|netcoreapp3.1|1.79μs|5.74ns|22.2ns|0.0125| 0|0|912&nbsp;B|
|#2602|`CallElasticsearch`|net472|2.85μs|15.5ns|94.4ns|0.125| 0|0|802&nbsp;B|
|#2602|`CallElasticsearch`|netcoreapp3.1|1.7μs|4.53ns|16.3ns|0.011| 0|0|792&nbsp;B|
|#2602|`CallElasticsearchAsync`|net472|2.99μs|13.2ns|47.6ns|0.148| 0.00159|0|939&nbsp;B|
|#2602|`CallElasticsearchAsync`|netcoreapp3.1|1.83μs|5.41ns|20.9ns|0.0125| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`ExecuteAsync`|net472|3.17μs|15.1ns|64.2ns|0.168| 0.00157|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`ExecuteAsync`|netcoreapp3.1|2.05μs|7.27ns|28.2ns|0.0147| 0|0|1.03&nbsp;KB|
|#2602|`ExecuteAsync`|net472|3.12μs|14.8ns|69.4ns|0.167| 0|0|1.06&nbsp;KB|
|#2602|`ExecuteAsync`|netcoreapp3.1|2.01μs|6.63ns|24.8ns|0.014| 0|0|1.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendAsync`|net472|6.83μs|22.3ns|83.5ns|0.36| 0|0|2.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendAsync`|netcoreapp3.1|4.83μs|17.6ns|68.2ns|0.0287| 0|0|2.21&nbsp;KB|
|#2602|`SendAsync`|net472|7μs|26ns|104ns|0.361| 0|0|2.28&nbsp;KB|
|#2602|`SendAsync`|netcoreapp3.1|4.95μs|23.5ns|96.9ns|0.0309| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|net472|3.35μs|10.4ns|39.1ns|0.227| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|netcoreapp3.1|3.12μs|9.97ns|38.6ns|0.0216| 0|0|1.53&nbsp;KB|
|#2602|`EnrichedLog`|net472|3.31μs|13.1ns|50.9ns|0.227| 0|0|1.45&nbsp;KB|
|#2602|`EnrichedLog`|netcoreapp3.1|3.07μs|9.93ns|38.5ns|0.0199| 0|0|1.53&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|net472|292μs|719ns|2.78μs|0.43| 0.143|0|4.33&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|netcoreapp3.1|230μs|753ns|2.92μs|0| 0|0|4.21&nbsp;KB|
|#2602|`EnrichedLog`|net472|289μs|1.14μs|4.42μs|0.429| 0.143|0|4.33&nbsp;KB|
|#2602|`EnrichedLog`|netcoreapp3.1|230μs|582ns|2.54μs|0| 0|0|4.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|net472|7.84μs|33.2ns|129ns|0.503| 0|0|3.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|netcoreapp3.1|6.74μs|39ns|321ns|0.0501| 0|0|3.6&nbsp;KB|
|#2602|`EnrichedLog`|net472|7.6μs|30ns|104ns|0.504| 0|0|3.23&nbsp;KB|
|#2602|`EnrichedLog`|netcoreapp3.1|6.56μs|37.3ns|269ns|0.0496| 0|0|3.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendReceive`|net472|2.37μs|12.7ns|65.9ns|0.16| 0.00118|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`SendReceive`|netcoreapp3.1|2.08μs|6.57ns|24.6ns|0.0143| 0|0|1.01&nbsp;KB|
|#2602|`SendReceive`|net472|2.34μs|9.89ns|38.3ns|0.16| 0.00119|0|1.01&nbsp;KB|
|#2602|`SendReceive`|netcoreapp3.1|2.06μs|6.95ns|26.9ns|0.0133| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|net472|6.03μs|21.1ns|81.8ns|0.291| 0|0|1.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`EnrichedLog`|netcoreapp3.1|5.3μs|21.1ns|78.9ns|0.0207| 0|0|1.49&nbsp;KB|
|#2602|`EnrichedLog`|net472|5.99μs|28ns|112ns|0.291| 0|0|1.87&nbsp;KB|
|#2602|`EnrichedLog`|netcoreapp3.1|5.17μs|13.3ns|47.8ns|0.0203| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`StartFinishSpan`|net472|913ns|3.84ns|14.9ns|0.072| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`StartFinishSpan`|netcoreapp3.1|875ns|2.27ns|7.85ns|0.00638| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`StartFinishScope`|net472|1.12μs|5.62ns|25.7ns|0.0843| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`StartFinishScope`|netcoreapp3.1|1.06μs|3.58ns|13.4ns|0.00781| 0|0|576&nbsp;B|
|#2602|`StartFinishSpan`|net472|908ns|4.77ns|23.4ns|0.072| 0|0|457&nbsp;B|
|#2602|`StartFinishSpan`|netcoreapp3.1|872ns|4.05ns|16.2ns|0.00634| 0|0|456&nbsp;B|
|#2602|`StartFinishScope`|net472|1.11μs|5.79ns|27.8ns|0.0841| 0|0|538&nbsp;B|
|#2602|`StartFinishScope`|netcoreapp3.1|1.03μs|5.74ns|35.8ns|0.00773| 0|0|576&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`RunOnMethodBegin`|net472|1.29μs|6.42ns|26.5ns|0.084| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9)|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|3.26ns|12.2ns|0.00786| 0|0|576&nbsp;B|
|#2602|`RunOnMethodBegin`|net472|1.26μs|3.29ns|11.8ns|0.0839| 0|0|538&nbsp;B|
|#2602|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|5.16ns|20ns|0.00814| 0|0|576&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2602 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9) will **not change** line coverage
:heavy_check_mark: Merging #2602 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9) will **not change** branch coverage
:no_entry: Merging #2602 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9) will will **increase** complexity by `173`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9) | #2602       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `12530` / `17525` | `12658` / `17761` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `7055` / `10227` | `7137` / `10385` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `11470`      | `11643`        |  `173`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=94418&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/94418/artifacts?artifactName=Code%20Coverage%20Report_94418&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2602](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=94602&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/94602/artifacts?artifactName=Code%20Coverage%20Report_94602&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/d0b92edd6204d5c9eb564b2075a0f4e778db93e9) | #2602       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `11470`      | `11643`       |  `173` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Agent.StreamFactories.TcpStreamFactory](https://github.com/DataDog/dd-trace-dotnet/tree/c718ac7d825da2eecd469f5b348eb350d9ec7693/tracer/src/Datadog.Trace/Agent/StreamFactories/TcpStreamFactory.cs) | `-90%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.JsonHttpClientTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/c718ac7d825da2eecd469f5b348eb350d9ec7693/tracer/src/Datadog.Trace/Telemetry/Transports/JsonHttpClientTelemetryTransport.cs) | `-26%` :no_entry: | `-38%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.TracesTransportStrategy](https://github.com/DataDog/dd-trace-dotnet/tree/c718ac7d825da2eecd469f5b348eb350d9ec7693/tracer/src/Datadog.Trace/Agent/TracesTransportStrategy.cs) | `-15%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.DiagnosticListeners.AspNetCoreDiagnosticObserver](https://github.com/DataDog/dd-trace-dotnet/tree/c718ac7d825da2eecd469f5b348eb350d9ec7693/tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs) | `-3%` :warning: | `-7%` :no_entry:   | `0` :heavy_check_mark: |


The following classes were added in #2602:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Ci.CodeOwners](https://github.com/DataDog/dd-trace-dotnet/pull/2602/files) | `62%` | `57%` | `69` |
| [Datadog.Trace.PDBs.MethodSymbolResolver](https://github.com/DataDog/dd-trace-dotnet/pull/2602/files) | `64%` | `68%` | `72` |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=94418&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/94418/artifacts?artifactName=Code%20Coverage%20Report_94418&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2602](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=94602&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/94602/artifacts?artifactName=Code%20Coverage%20Report_94602&api-version=6.0&%24format=zip)

File: tracer/src/Datadog.Trace.ClrProfiler.Native/rejit_handler.cpp [hunk 1]
```diff
@@ -356,51 +356,59 @@ void RejitHandler::RemoveModule(ModuleID moduleId)
...
+        {
+            mod.second->RequestRejitForInlinersInModule(moduleInliner);
         }
     }
 
+    // If the module is not in the inliners list we added and request rejit for it.
     if (!alreadyAdded)
     {
         // Add the new module inliner
         m_ngenInlinersModules.push_back(moduleId);
```
Comment (zacharycmontoya) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/rejit_handler.cpp:407]: There's a behavioral difference here: When `!alreadyLoaded && m_profilerInfo == nullptr`, we no longer execute this line. Is that okay?

**Review by zacharycmontoya — APPROVED** [review]

File: tracer/src/Datadog.Trace.ClrProfiler.Native/rejit_handler.cpp (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/rejit_handler.cpp:407]: yes, that was intentional, it doesn't make sense to do anything if we don't have the profilerInfo interface.

**Review by kevingosse — APPROVED** [review]

**Pull request merged**